src/coreclr/tools/dotnet-pgo/PgoRootCommand.cs

```diff
@@ -0,0 +1,297 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Help;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+
+namespace Microsoft.Diagnostics.Tools.Pgo
+{
+    internal sealed class PgoRootCommand : RootCommand
+    {
+        public Option<string[]> InputFilesToMerge { get; } =
+            new(new[] { "--input", "-i" }, "Input .mibc files to be merged. Multiple input arguments are specified as --input file1.mibc --input file2.mibc") { IsRequired = true, Arity = ArgumentArity.OneOrMore };
+        public Option<string[]> InputFilesToCompare { get; } =
+            new(new[] { "--input", "-i" }, "The input .mibc files to be compared. Specify as --input file1.mibc --input file2.mibc") { IsRequired = true, Arity = new ArgumentArity(2, 2) /* exactly two */ };
+        public Option<string> InputFileToDump { get; } =
+            new(new[] { "--input", "-i" }, "Name of the input mibc file to dump") { IsRequired = true, Arity = ArgumentArity.ExactlyOne };
+        public Option<string> TraceFilePath { get; } =
+            new(new[] { "--trace", "-t" }, "Specify the trace file to be parsed");
+        public Option<string> OutputFilePath { get; } =
+            new(new[] { "--output", "-o" }, "Specify the output filename to be created");
+        public Option<string> PreciseDebugInfoFile { get; } =
+            new(new[] { "--precise-debug-info-file" }, "Name of file of newline separated JSON objects containing precise debug info");
+        public Option<int> Pid { get; } =
+            new(new[] { "--pid" }, "The pid within the trace of the process to examine. If this is a multi-process trace, at least one of --pid or --process-name must be specified");
+        public Option<string> ProcessName { get; } =
+            new(new[] { "--process-name" }, "The process name within the trace of the process to examine. If this is a multi-process trace, at least one of --pid or --process-name must be specified");
+        public Option<Dictionary<string, string>> Reference =
+            new(new[] { "--reference", "-r" }, result => Helpers.BuildPathDictionay(result.Tokens, false), true, "If a reference is not located on disk at the same location as used in the process, it may be specified with a --reference parameter. Multiple --reference parameters may be specified. The wild cards * and ? are supported by this option");
```

Comment:
In fact, other options that were using `DefineFileOptionList` also supported wild cards. For example, the inputs given to `merge`. I've not personally used it for anything but `--reference`, but I'm not sure if it is ok to remove the support. cc @davidwrighton